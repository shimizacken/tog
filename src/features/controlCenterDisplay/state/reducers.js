const initialState = {
    drivingInfo: '',
    warningInfo: ''
};

const controlCenterDisplay = (state = initialState, action) => {
    return state;
};

export { controlCenterDisplay };