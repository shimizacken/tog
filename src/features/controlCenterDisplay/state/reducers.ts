import type { ControlCenterDisplayState } from "../../../state/store.types";

const initialState: ControlCenterDisplayState = {
  drivingInfo: "",
  warningInfo: "",
};

export const controlCenterDisplay = (state = initialState) => {
  return state;
};
