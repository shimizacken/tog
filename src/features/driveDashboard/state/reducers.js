const DrivingButtonStates = {
    TURNED_OFF: 0,
    READY: 1,
    DRIVE: 2,
    ERROR: 3
};

const initialState = {
    drivingButtonState: DrivingButtonStates.READY,
    throttleStick: 0
};

const driveDashboard = (state = initialState, action) => {
    return state;
 };

 export { driveDashboard };