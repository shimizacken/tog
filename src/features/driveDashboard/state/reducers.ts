import {
  SET_DRIVE_BUTTON_STATUS,
  SET_THROTTLE_SPEED,
  SET_DRIVE_BUTTON_SWITCHING_STATUS,
  SET_DRIVING_STATUS,
} from "./constants";
import { DrivingButtonStates } from "../services/ui/enums/drivingButtonStates";
import {
  SetDrivingButtonState,
  SetDrivingButtonSwitchingState,
  SetDrivingStatus,
} from "./actions";
import { Action } from "redux";
import type { DriveDashboardState } from "../../../state/store.types";

const driveDashboardInitialState: DriveDashboardState = {
  drivingButtonState: DrivingButtonStates.TURNED_OFF,
  throttleSpeed: 0,
  drivingStatus: false,
};

export const driveDashboard = (
  state = driveDashboardInitialState,
  action: Action
) => {
  if (action.type === SET_DRIVE_BUTTON_STATUS) {
    return {
      ...state,
      drivingButtonState: (action as SetDrivingButtonState).drivingButtonState,
    };
  }

  if (action.type === SET_DRIVE_BUTTON_SWITCHING_STATUS) {
    return {
      ...state,
      drivingButtonState: (action as SetDrivingButtonState).drivingButtonState,
    };
  }

  if (action.type === SET_THROTTLE_SPEED) {
    return {
      ...state,
      throttleSpeed: (action as SetDrivingButtonSwitchingState).throttleSpeed,
    };
  }

  if (action.type === SET_DRIVING_STATUS) {
    return {
      ...state,
      drivingStatus: (action as SetDrivingStatus).drivingStatus,
    };
  }

  return state;
};
