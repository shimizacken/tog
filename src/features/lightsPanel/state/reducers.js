const LightStates = {
    OFF: 'OFF',
    ON: 'ON',
    ERROR: 'ERROR'
};

const lightsInitialState = {
    frontLightsState: LightStates.ON,
    backLightsState: LightStates.ON,
    cabinLights: LightStates.ON,
    wagonsLights: LightStates.ON
};

const lights = (state = lightsInitialState, action) => {
    return state;
};

export { lights };