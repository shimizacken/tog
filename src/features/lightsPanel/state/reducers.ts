import { Action } from "redux";
import {
  SET_FRONT_LIGHT_STATE,
  SET_BACK_LIGHT_STATE,
  SET_CABIN_LIGHT_STATE,
  SET_WAGON_LIGHT_STATE,
} from "./constants";

export type LightsState = {
  readonly frontLightsState: false;
  readonly backLightsState: false;
  readonly cabinLightsState: false;
  readonly wagonsLightsState: false;
};

const lightsInitialState: LightsState = {
  frontLightsState: false,
  backLightsState: false,
  cabinLightsState: false,
  wagonsLightsState: false,
};

export const lights = (state = lightsInitialState, action: Action) => {
  if (action.type === SET_FRONT_LIGHT_STATE) {
    return {
      ...state,
      frontLightsState: !state.frontLightsState,
    };
  }

  if (action.type === SET_BACK_LIGHT_STATE) {
    return {
      ...state,
      backLightsState: !state.backLightsState,
    };
  }

  if (action.type === SET_CABIN_LIGHT_STATE) {
    return {
      ...state,
      cabinLightsState: !state.cabinLightsState,
    };
  }

  if (action.type === SET_WAGON_LIGHT_STATE) {
    return {
      ...state,
      wagonsLightsState: !state.wagonsLightsState,
    };
  }

  return state;
};
