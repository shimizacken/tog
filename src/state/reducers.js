import { combineReducers } from 'redux';
import { driveDashboard } from '../features/driveDashboard/state/reducers';
import { controlCenterDisplay } from '../features/controlCenterDisplay/state/reducers';

const reducers = combineReducers({
    driveDashboard,
    controlCenterDisplay
});

export default reducers;
