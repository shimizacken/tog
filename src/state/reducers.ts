import { combineReducers } from "redux";
import { driveDashboard } from "../features/driveDashboard/state/reducers";
import { controlCenterDisplay } from "../features/controlCenterDisplay/state/reducers";
import { lights } from "../features/lightsPanel/state/reducers";

export const reducers = combineReducers({
  driveDashboard,
  controlCenterDisplay,
  lights,
});
