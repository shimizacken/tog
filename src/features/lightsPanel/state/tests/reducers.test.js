import { lights } from "../reducers";


describe('lights reducer test suite', () => {
  
    it('should return initial state', () => {
        
        expect(lights()).toMatchSnapshot();
    });
});
