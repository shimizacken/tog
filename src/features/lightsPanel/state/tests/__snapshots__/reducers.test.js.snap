// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lights reducer test suite should return initial state 1`] = `
Object {
  "backLightsState": "ON",
  "cabinLights": "ON",
  "frontLightsState": "ON",
  "wagonsLights": "ON",
}
`;
